aws/cloudformation/cloud_formation_stack.yml.erb

```diff
@@ -186,6 +186,10 @@ Resources:
           - Effect: Allow
             Action: 's3:*'
             Resource: '*'
+          # SageMaker Invoke Endpoint access
+          - Effect: Allow
+            Action: 'sagemaker:InvokeEndpoint'
+            Resource: '*'
```

Comment:
Can we constrain this permission to just the Endpoint (or set of Endpoints) that we’ve provisioned for this feature?

[This documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/api-permissions-reference.html) indicates that the AWS permissions system (“IAM”) supports restricting to specific Endpoints by their unique AWS identifier (“ARN”) or through wildcards, a set of Endpoints.

I think we could specify something like below to permit invocation an Endpoint named `generativeAITutorialStaging` and an Endpoint named `generativeAITutorialProduction`

`Resource: !Sub “ arn:aws:sagemaker: ${AWS::Region}:${AWS::AccountId}:endpoint/generativeAITutorial*”`

Reply:
And, yes, this would require creating an endpoint with a different naming convention than `mistral-7b-inst-self-hosted`

Reply:
That's fine with me. As long as we can use some common prefix we should be good from a testing and piloting perspective. How does something like `endpoint/gen-ai-tutorial-*` sound?

Reply:
That naming makes sense to me.